src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -758,6 +789,22 @@ ExecuteUtilityTaskListWithoutResults(List *taskList)
 }
 
 
+/*
+ * ExecuteTaskListRepartiton is a proxy to ExecuteTaskListExtended() with defaults
+ * for some of the arguments for a repartition query.
+ */
+uint64
+ExecuteTaskListRepartition(RowModifyLevel modLevel, List *taskList, int targetPoolSize)
```

Comment:
Maybe move this into the re-partitioning-specific file, or give it a more generic name.

Reply:
`ExecuteTaskListOutsideTransaction` seems like a better name